samples/gfx_html/mongoose.cpp

```diff
@@ -2372,7 +2372,7 @@ int mg_url_decode(const char *src, int src_len, char *dst,
 #define HEXTOI(x) (isdigit(x) ? x - '0' : x - 'W')
 
   for (i = j = 0; i < src_len && j < dst_len - 1; i++, j++) {
-    if (src[i] == '%' && i < src_len - 2 &&
+    if (i < src_len - 2 && src[i] == '%' &&
```

Comment:
This is also a question of style. src[i] will always be valid because of check if 'for' statement

Reply:
In for you compare i with src_len, in if you compare i with src_len  - 2, so it’s not only style

Reply:
I mean that there will be no out of bounds exception. So these two statements are equal in performance.